File: exercise/week02/c#/Shift/Vertical.cs
namespace Shift;

public static class Vertical
{
    public static int WhichFloor(string signalStream)
    {
        List<Tuple<char, int>> val = [];

        for (int i = 0; i < signalStream.Length; i++)
        {
            var c = signalStream[i];

            if (signalStream.Contains("🧝"))
            {
                int j;
                if (c == ')') j = 3;
                else j = -2;

                val.Add(new Tuple<char, int>(c, j));
            }
            else if (!signalStream.Contains("🧝"))
            {
                val.Add(new Tuple<char, int>(c, c == '(' ? 1 : -1));
            }
            else val.Add(new Tuple<char, int>(c, c == '(' ? 42 : -2));
        }

        int result = 0;
        foreach (var kp in val)
        {
            result += kp.Item2;
        }
        return result;
    }
}

File: exercise/week02/php/src/Shift/Vertical.php
<?php

namespace Shift;

class Vertical
{
    public static function whichFloor(string $signalStream): int
    {
        $val = [];

        for ($i = 0; $i < strlen($signalStream); $i++) {
            $c = $signalStream[$i];

            if (strpos($signalStream, "🧝")) {
                $j = ($c === ')') ? 3 : -2;
                $val[] = [$c, $j];
            } else if (strpos($signalStream, "🧝") === false) {
                $val[] = [$c, ($c === '(') ? 1 : -1];
            } else {
                $val[] = [$c, ($c === '(') ? 42 : -2];
            }
        }

        $result = 0;
        foreach ($val as $kp) {
            $result += $kp[1];
        }

        return $result;
    }
}